transcode/transcode.go

```diff
@@ -411,6 +412,23 @@ func transcodeSegment(
 ) error {
 	start := time.Now()
 
+	transcodeConf := clients.LivepeerTranscodeConfiguration{
+		TimeoutMultiplier: 10,
+	}
+	// If this is a request to transcode a Clip source input, then
+	// force T to do a re-init of transcoder after segment at idx=0.
+	// This is required because the segment at idx=0 is a locally
+	// re-encoded segment and the following segment at idx=1 is a
+	// source recorded segment. Without a re-init of the transcoder,
+	// the different encoding between the two segments causes the
+	// transcode operation to incorrectly tag the output segment as
+	// having two video tracks.
+	if transcodeRequest.IsClip && int64(segment.Index) == 1 {
+		transcodeConf.ForceSessionReinit = true
+	} else {
+		transcodeConf.ForceSessionReinit = false
```

Comment:
Nit: I think you can skip this `else` since `false` is the default value of bool.

Reply:
Or just `transcodeConf.ForceSessionReinit = transcodeRequest.IsClip && int64(segment.Index) == 1`

Reply:
Do we not need a re-init for the final segment too?

Reply:
> Do we not need a re-init for the final segment too (when that's also clipped)?

That's a good point. I guess that yes!

Reply:
> Do we not need a re-init for the final segment too (when that's also clipped)?
No actually, surprisingly. For some reason, the transcoder session is able to handle going from source recording segment to final segment that's also clipped (i.e re-encoded differently). Need to understand lpms better to understand why that is. Planning to follow up with clipping directly at T layer and hopefully understand this better.